src/main/scala/com/cognite/sdk/scala/common/OAuth2.scala

```diff
@@ -91,7 +90,7 @@ object OAuth2 {
             .response(asJson[ClientCredentialsResponse])
             .send(sttpBackend)
           payload <- response.body match {
-            case Right(response) => F.pure(response)
+            case Right(response) => F.delay(response)
```

Comment:
No big deal, but I'd think this is definitely pure?

Reply:
done :)